packages/loader/container-loader/src/retriableDocumentStorageService.ts

```diff
@@ -35,6 +45,26 @@ export class RetriableDocumentStorageService extends DocumentStorageServiceProxy
         return this.readWithRetry(async () => this.internalStorageService.readBlob(id));
     }
 
+    public async getVersions(versionId: string, count: number): Promise<IVersion[]> {
```

Comment:
I'm a bit torn on that one. On one hand it's nice to have it wrapped here (and it's required for calls coming from summarizer). On another hand, that will make container load a bit weird, because there is no way for user to cancel it (we are too early in Container.load and host does not have access to container object yet).
I think that's fine and we should optimize work to return empty container sooner (before any load), but something to keep in mind

Reply:
Yes we already have a issue opened for that as same concern is true for getSnapshotTree call but we already fetch the snapshot while making the getVersions call. So yes  we will address it by finding a logic on returnig container earlier.